### ISSUE TITLE
num_prototype

### ISSUE THREAD

**Issue by txw1997** [facebookresearch/swav#32]
Hi, thanks for your excellent work! 
I have a question about num_prototype in deepclustering V2. What does the num_prototype mean?
why num_prototype can bigger than class_num?
Thanks!

**Comment by txw1997** [comment]
And, although my loss not stuck in a fixed value, but there was no obvious decline overall.

**Comment by mathildecaron31** [comment]
Hi, thanks for you interest in this work. In DeepCluster-v2 the num_prototypes argument design the number of clusters used for k-means clustering. As our method is unsupervised there is no notion of class.

Can you provide more details about the loss ? It should be decreasing along training. If not, it might be that the hyperparameters are not set properly.

**Comment by davidsonic** [comment]
I'm wondering at what level should the loss be will the training be considered complete?

**Comment by txw1997** [comment]
Understand, thanks for you reply. I follow the  hyperparameters in other issue, and the loss become normal. Thanks!!
So, the clustering is aim to learn a good embedding for a image, and then use a classifier to classify them, am i right?

**Comment by mathildecaron31** [comment]
@txw1997 yes exactly. Clustering is a proxy task to pre-train the network without using labels. Then we can use this pre-trained network to solve downstream supervised tasks like classification for example.

**Comment by mathildecaron31** [comment]
@davidsonic in my experiments I've observed that the loss usually decreases between 3 and 4.

For example see the training loss curve for our 800 epochs training:
![image](https://user-images.githubusercontent.com/20599397/96837186-4d2d1b00-1446-11eb-9495-a679321d1554.png)

**Comment by davidsonic** [comment]
Cool, thanks for the information!

**Comment by mathildecaron31** [comment]
I'm closing the issue. Feel free to reopen if you need further assistance.
